app/lib/api.py

```diff
@@ -1,48 +1,67 @@
-import requests
+import logging
 
-# from flask import current_app
+from requests import JSONDecodeError, Timeout, TooManyRedirects, codes, get
 
+logger = logging.getLogger(__name__)
 
-class ApiResourceNotFound(Exception):
+
+class ResourceNotFound(Exception):
     pass
 
 
-class BaseAPI:
+class JSONAPIClient:
     api_url = ""
-    api_path = "/"
     params = {}
 
-    def __init__(self, api_url):
+    def __init__(self, api_url, params={}):
         self.api_url = api_url
+        self.params = params
 
     def add_parameter(self, key, value):
         self.params[key] = value
 
-    def get_results(self, page=None):
-        if page:
-            self.add_parameter("page", page)
-        url = f"{self.api_url}{self.api_path}"
-        # current_app.logger.debug(
-        #     f"API endpoint requested: {url} (params {self.params})"
-        # )
+    def add_parameters(self, params):
+        self.params = self.params | params
+
+    def get(self, path="/"):
+        url = f"{self.api_url}/{path.lstrip('/')}"
+        headers = {
+            "Cache-Control": "no-cache",
+            # "Accept": "application/json",  # TODO: This breaks the API
+        }
         try:
-            response = requests.get(url, params=self.params)
+            response = get(
```

Comment:
Why using `requests` - `get()` over `sessions.get()` as in `ds-wagtail` ?

Reply:
In production, we could have 2-4 images and they all need to be aware of the same sessions as each container should be stateless. Until we have persistent session handling across multiple container instances, this is not going to be possible so I am just using the bare `request` to get the query params.